lib/k8s/resource_client.rb

```diff
@@ -224,20 +224,20 @@ def watch(labelSelector: nil, fieldSelector: nil, resourceVersion: nil, timeout:
       @transport.request(
         method: method,
         path: path,
-        response_class: K8s::API::MetaV1::List,
         query: make_query(
           'labelSelector' => selector_query(labelSelector),
           'fieldSelector' => selector_query(fieldSelector),
           'resourceVersion' => resourceVersion,
-          'watch' => '1'
+          'watch' => '1',
+          'timeoutSeconds' => timeout
         ),
         response_block: lambda do |chunk, _, _|
           data = JSON.parse(chunk)
           event = K8s::API::MetaV1::WatchEvent.new(data)
           resourceVersion = event.resource&.metadata&.resourceVersion
           yield event
         end,
-        read_timeout: timeout
+        timeout: nil
```

Comment:
I think this may be an invalid (=ignored) option.

Reply:
(should probably be read_timeout or just nothing)

Reply:
`nil` makes excon not to timeout ever. If we don't pass `nil` then default read-timeout value is used.

Reply:
I can't find :timeout from excon sources, so this should probably be `read_timeout`

Reply:
Oh yes, good catch.